src/Divider.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+
+
+class Divider extends React.Component {
+
+    constructor () {
+      super(...arguments);
+  
+      this.state = {
+        dragging: false,
+        initPos: {x:null,y:null},
+      };
+    }
+  
+    // Add/remove event listeners based on drag state
+    componentDidUpdate (props, state) {
+      if (this.state.dragging && !state.dragging) {
+        document.addEventListener('mousemove', this.onMouseMove)
+        document.addEventListener('mouseup', this.onMouseUp)
+      } else if (!this.state.dragging && state.dragging) {
+        document.removeEventListener('mousemove', this.onMouseMove)
+        document.removeEventListener('mouseup', this.onMouseUp)
+      }
+    }
+  
+    // Start drag state and set initial position
+    onMouseDown = (e) => {
+  
+      // only left mouse button
+      if (e.button !== 0) return
+  
+      this.setState({
+        dragging: true,
+        initPos: {
+          x: e.pageX,
+          y: e.pageY
+        },
+      })
+  
+      e.stopPropagation()
+      e.preventDefault()
+    };
+  
+    // End drag state
+    onMouseUp = (e) => {
+      this.setState({dragging: false})
+      e.stopPropagation()
+      e.preventDefault()
+  
+      return this.onResizeComplete(this.props.panelID)
```

Comment:
Aside from this new function, were there any other changes two Divider or Panel - I can't easily see the changes since you separated them into their own files